idseq_dag/steps/run_lzw.py

```diff
@@ -1,10 +1,29 @@
 from idseq_dag.engine.pipeline_step import PipelineStep
 import idseq_dag.util.command as command
+from command import run_in_subprocess
 import idseq_dag.util.log as log
 import idseq_dag.util.count as count
+import idseq_dag.util.fasta as fasta
+from idseq_dag.util.thread_with_result import ThreadWithResult, execute_all
+from multiprocessing import cpu_count
+from typing import Iterator
+
 
 class PipelineStepRunLZW(PipelineStep):
 
+    MAX_SUBPROCS = 16
+
+    # Core count caveats:
+    #
+    #   * Due to hyperthreading, the core count is exagerated 2x.
+    #
+    #   * When running inside a docker container, cpu_count reports the number of
+    #     virtual CPU cores on the instance that is hosting the container.  There
+    #     could be limits preventing the container from using all those cores.
+    REAL_CORES = (cpu_count() + 1) // 2
```

Comment:
Why + 1?
Cant this script make use of hyperthreading?

Reply:
The `+1` is helpful if this should happen to run on an instance with just 1 vCPU.

Running on 2x the number of physical cores rarely produces a win for CPU-bound tasks, but there are costs for exrta processes, particularly in this case if they fall behind when CPU is oversubscribed their sliding window through the shared huge input file would fracture into multiple windows and potentially fragment the IO.

Reply:
In this case especially, each of the processes will parse the entire file, and while that's much faster than LZW, it's still not a trivial amount of work, and only comes for free if we do not oversubscribe the cores.

Reply:
Yeah, i guess i was missing the info that this was "CPU-bound".  Thanks for clarifying.